src/main/java/de/tum/in/www1/artemis/service/CourseScoreCalculationService.java

```diff
@@ -181,7 +306,7 @@ private double calculatePointsAchievedFromExercise(Exercise exercise, Result res
         return pointsAchievedFromExercise;
     }
 
-    Result getResultForParticipation(Participation participation, ZonedDateTime dueDate) {
+    private Result getResultForParticipation(Participation participation, ZonedDateTime dueDate) {
```

Comment:
Optional: I'd advise exploring if using the `ParticipantScore` to get the latest rated result speeds this method up. It should be faster than fetching the Participation and iterating over all Results to find the latest.

Reply:
Thanks for having a look! Do you think it would still make more sense to use the `ParticipantScore` if I do not have to fetch the participations (see Stephan's comment below)?